backend/layers/business/business.py

```diff
@@ -972,17 +972,8 @@ def publish_collection_version(
             if canonical_datasets != version_datasets:
                 has_dataset_revisions = True
 
-            # Update publisher metadata if there are updates in Crossref that have occurred since last publish.
-            doi_link = next((link for link in version.metadata.links if link.type == "DOI"), None)
-            if doi_link:
-                # Don't track/raise errors here; just keep the existing publisher metadata.
-                publisher_metadata, _, deposited_at_timestamp = self._get_publisher_metadata(doi_link.uri, [])
-                if (
-                    publisher_metadata
-                    and deposited_at_timestamp
-                    and datetime.fromtimestamp(deposited_at_timestamp) > date_of_last_publish
-                ):
-                    self.database_provider.save_collection_publisher_metadata(version_id, publisher_metadata)
+            # Check Crossref for updates in publisher metadata since last publish.
+            self._update_revision_publisher_metadata(version, date_of_last_publish)
```

Comment:
or update_crossref_metadata to be more generic

Reply:
Hey @nayib-jose-gloria, double-checking the requirement here as I was under the impression this check is specific to revisions. The [Upon Publishing a Revision, Pull Any Updates to DOI metadata from Crossref API](https://github.com/chanzuckerberg/single-cell-data-portal/issues/6924) ticket states:

> If a collection version is a revision

Reply:
it's more pertinent for revisions because those are the ones likely to have a DOI change over time (vs. a short-lived initial private collection). But `publish_collection_version` is run on both initial private collections and revisions, so it seems reasonable to cover both cases since it has to go through this code path regardless. I don't believe it would require any logical changes

Reply:
Got it. The call to `_update_revision_publisher_metadata` is currently nested in an "if revision" block so I'll pull it out of there so it is always executed. Two follow-on questions:

1. Should I compare the Crossref `deposited` date against the collection version `created_at` for private collections?
2. Can I go ahead and update the ticket with these updated requirements?

Reply:
1. Not required, but a good idea if you have time to go ahead and implement that. 
2. Sure